persistence/memorypersistence/aggregate.go

```diff
@@ -46,23 +64,61 @@ func (v *validator) VisitSaveAggregateMetaData(
 	}
 }
 
+// VisitSaveAggregateSnapshot returns an error if a "SaveAggregateSnapshot"
+// operation can not be applied to the database.
+func (v *validator) VisitSaveAggregateSnapshot(
+	_ context.Context,
+	op persistence.SaveAggregateSnapshot,
+) error {
+	ss := op.Snapshot
+
+	if ss.HandlerKey != "" && ss.InstanceID != "" {
+		return nil
+	}
+
+	return persistence.NotFoundError{
+		Cause: op,
+	}
```

Comment:
The `validator` need only verify that it is semantically ok to apply this operation to the database, it doesn't need to perform general import validation.  Because it's always ok to save a snapshot, this can simply return `nil`.

```suggestion
	return nil
```